source/dub/project.d

```diff
@@ -643,16 +643,33 @@ class Project {
 	/// Returns a map with the configuration for all packages in the dependency tree.
 	string[string] getPackageConfigs(in BuildPlatform platform, string config, bool allow_non_library = true)
 	const {
+		import std.typecons : Rebindable, rebindable;
+
 		struct Vertex { string pack, config; }
 		struct Edge { size_t from, to; }
 
 		Vertex[] configs;
 		Edge[] edges;
 		string[][string] parents;
 		parents[m_rootPackage.name] = null;
-		foreach (p; getTopologicalPackageList())
-			foreach (d; p.getAllDependencies())
-				parents[d.name.toString()] ~= p.name;
+		// cached package information to avoid continuous re-computation
+		// during the resolution process
+		const(Package)[] package_list;
+		Rebindable!(const(Package))[string] package_map;
+		string[const(Package)] package_names;
```

Comment:
How is this faster/better than `package.name`?

Reply:
As the commit message states, `package.name` needs to allocate for every call. In particular, it needs to construct the full name from the parent package name concatenated with '":"' and the shallow name of the sub package. Also note that the final version of the code uses a linear array instead of an associative array.